src/Tools/BuildBoss/StructuredLoggerChecekerUtil.cs

```diff
@@ -0,0 +1,148 @@
+    /// is here:
+    /// 
+    ///     https://github.com/KirillOsenkov/MSBuildStructuredLog/issues/42
+    /// </summary>
+    internal sealed class StructuredLoggerChecekerUtil : ICheckerUtil
+    {
+        internal static readonly StringComparer FilePathComparer = StringComparer.OrdinalIgnoreCase;
+        private static readonly string s_copyingFileFrom = "Copying file from \"";
+        private static readonly string s_creatingHardLink = "Creating hard link to copy \"";
+        private static readonly string s_didNotCopy = "Did not copy from file \"";
+        private static readonly string s_to = "\" to \"";
+
+        private readonly XDocument _document;
+        private readonly Dictionary<string, List<string>> _copyMap = new Dictionary<string, List<string>>(FilePathComparer);
+
+        internal StructuredLoggerChecekerUtil(XDocument document)
+        {
+            _document = document;
+        }
+
+        public bool Check(TextWriter textWriter)
+        {
+            foreach (var element in _document.XPathSelectElements("//CopyTask"))
+            {
+                foreach (var message in element.Elements(XName.Get("Message")))
+                {
+                    ProcessMessage(message);
+                }
+            }
+
+            var allGood = true;
+            foreach (var pair in _copyMap.OrderBy(x => x.Key))
+            {
+                var list = pair.Value;
+                if (list.Count > 1)
+                {
+                    textWriter.WriteLine($"Multiple writes to {pair.Key}");
+                    foreach (var item in list)
+                    {
+                        textWriter.WriteLine($"\t{item}");
+
+                    }
+                    textWriter.WriteLine("");
+
+                    allGood = false;
+                }
+            }
+
+            return allGood;
+        }
+
+        private void ProcessMessage(XElement element)
+        {
+            var text = element.Value.Trim();
+            if (text.StartsWith(s_copyingFileFrom))
+            {
+                ProcessCopyingFileFrom(text, s_copyingFileFrom, s_to);
+            }
+            else if (text.StartsWith(s_creatingHardLink))
+            {
+                ProcessCopyingFileFrom(text, s_creatingHardLink, s_to);
+            }
+            else if (text.StartsWith(s_didNotCopy))
+            {
+                // Ignore files which were not copied.  This logic comes from the original 
+                // analysis.
+            }
+        }
+
+        private void ProcessCopyingFileFrom(string text, string prefix, string infix)
+        {
+            var split = text.IndexOf(infix);
+            var prefixLength = prefix.Length;
+            int toLength = infix.Length;
+            var source = text.Substring(prefixLength, split - prefixLength);
+            var destination = text.Substring(split + toLength, text.Length - 2 - split - toLength);
```

Comment:
What is the `-2` for?

This seems likely to crash when a new string format appears from msbuild.